telemetry/prometheus/src/main/scala/io/buoyant/telemetry/prometheus/PrometheusTelemeter.scala

```diff
@@ -34,14 +34,21 @@ class PrometheusTelemeter(metrics: MetricsTree) extends Telemeter with Admin.Wit
   def tracer = NullTracer
   def run(): Closable with Awaitable[Unit] = Telemeter.nopRun
 
-  private[this] val disallowedChars = "[^a-zA-Z0-9:]".r
-  private[this] def escapeKey(key: String) = disallowedChars.replaceAllIn(key, "_")
+  private[this] val metricNameDisallowedChars = "[^a-zA-Z0-9:]".r
+  private[this] def escapeKey(key: String) = metricNameDisallowedChars.replaceAllIn(key, "_")
+
+  private[this] val labelKeyDisallowedChars = "[^a-zA-Z0-9_]".r
+  private[this] def escapeLabelKey(key: String) = labelKeyDisallowedChars.replaceAllIn(key, "_")
+
+  // https://prometheus.io/docs/instrumenting/exposition_formats/#text-format-details
+  private[this] val labelValDisallowedChars = """(\\|\"|\n)""".r
+  private[this] def escapeLabelVal(key: String) = labelValDisallowedChars.replaceAllIn(key, """\\\\""")
```

Comment:
yeah - we want to replace any `\` with `\\`, and each `\` here has to be escaped, even in the triple quotes. the backslash character in the replace string is still interpreted as an escape character...! 
https://www.scala-lang.org/api/current/scala/util/matching/Regex.html

```
scala> ("abc".r).replaceAllIn("abcdef", """\\""")
res106: String = \def

scala> ("abc".r).replaceAllIn("abcdef", """\\\\""")
res107: String = \\def

scala> ("abc".r).replaceAllIn("abcdef", """\""")
java.lang.IllegalArgumentException: character to be escaped is missing
  at java.util.regex.Matcher.appendReplacement(Matcher.java:809)
  at java.util.regex.Matcher.replaceAll(Matcher.java:955)
  at scala.util.matching.Regex.replaceAllIn(Regex.scala:450)
  ... 33 elided
```